tests/lib/proc_utils.py

```diff
@@ -5,11 +5,29 @@
 """ Common utilities for working with processes.
 """
 
-import logging
 import subprocess
+import sys
+import logging
+
+
+def get_env_from(script):
+    """ Get environment set by a shell script
+    """
+    env = {}
+    if sys.platform == "win32":
+        cmd = f'"{script}" && set'
+    else:
+        cmd = f'source "{script}" && env'
+    dump = subprocess.check_output(cmd, shell=True, universal_newlines=True).strip()
+    for line in dump.split("\n"):
+        # split by first '='
+        pair = [str(val).strip() for val in line.split("=", 1)]
+        if len(pair) > 1 and pair[0] and pair[1] is not None:  # ignore invalid entries
```

Comment:
I'm not sure `pair[0] and pair[1] is not None` means `pair[0] is not None and pair[1] is not None`. Should it be extended this way?

Reply:
And may str() be equal to `None`? Looks like the check is redundant

Reply:
pair[0] should not be empty string or none
pair[1] can be empty string but should not be none

Reply:
`pair` is list of 1 or 2 strings (since .split(char, 1) and `str(...)` conversion), so I see no reason to compare with None because it's always True